test/runtests.jl

```diff
@@ -1,44 +1,54 @@
 using TestReports
-using Base.Test
+using Test
 using ReferenceTests
 
-"Strip the filenames from the string, so that the reference strings work on different computers"
-strip_filepaths(str) = replace(str, r" at .*\d+$"m, "")
+# Strip the filenames from the string, so that the reference strings work on different computers
+strip_filepaths(str) = replace(str, r" at .*\d+$"m => "")
+
+# Replace direction of windows slashes so reference strings work on windows
+replace_windows_filepaths(str) = replace(str, ".\\" => "./")
+
+# Replace Int32s so reference strings work on x86 platforms
+replace_Int32s(str) = replace(str, "Int32" => "Int64")
 
 @testset "SingleNest" begin
-    @test_reference "references/singlenest.txt" readstring(`$(Base.julia_cmd()) -e "using Base.Test; using TestReports; (@testset ReportingTestSet \"blah\" begin @testset \"a\" begin @test 1 ==1 end end) |> report |> print"`) |> strip_filepaths
+    @test_reference "references/singlenest.txt" read(`$(Base.julia_cmd()) -e "using Test; using TestReports; (@testset ReportingTestSet \"blah\" begin @testset \"a\" begin @test 1 ==1 end end) |> report |> print"`, String) |> strip_filepaths |> replace_windows_filepaths |> replace_Int32s
 end
 
 @testset "Complex Example" begin
-    @test_reference "references/complexexample.txt" readstring(`$(Base.julia_cmd()) $(@__DIR__)/example.jl`) |> strip_filepaths
+    if VERSION >= v"1.4.0"
+        @test_reference "references/complexexample.txt" read(`$(Base.julia_cmd()) $(@__DIR__)/example.jl`, String) |> strip_filepaths |> replace_windows_filepaths |> replace_Int32s
+    else
+        @test_reference "references/complexexample_julia_1_3.txt" read(`$(Base.julia_cmd()) $(@__DIR__)/example.jl`, String) |> strip_filepaths |> replace_windows_filepaths |> replace_Int32s
+    end
 end
 
 
 @testset "any_problems" begin
-    
+
     fail_code = """
-    using Base.Test
+    using Test
     using TestReports
     ts = @testset ReportingTestSet "eg" begin
         @test false == true
     end;
     exit(any_problems(ts))
     """
-    
-    @test_throws ErrorException run(`$(Base.julia_cmd()) -e $(fail_code)`)
-    
-     
+
+    @test_throws ProcessFailedException run(`$(Base.julia_cmd()) -e $(fail_code)`)
```

Comment:
```suggestion
    @test_throws Exception run(`$(Base.julia_cmd()) -e $(fail_code)`)
```

Depending on julia version this can be a `ErrorException` or a `ProcessFailedException`
and we really don't care which just as long as it was a failing exit code